staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -191,3 +192,7 @@ type namedCreaterAdapter struct {
 func (c *namedCreaterAdapter) Create(ctx context.Context, name string, obj runtime.Object, createValidatingAdmission rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	return c.Creater.Create(ctx, obj, createValidatingAdmission, options)
 }
+
+func prefixFromUserAgent(u string) string {
+	return strings.Split(u, "/")[0]
```

Comment:
We're sure this is useful and doesn't return "kubernetes" for everything?

Reply:
Some examples of values we've seen:
`kubelet`, `hyperkube`, `kubectl`, ...